core/src/test/scala/unit/kafka/log/remote/RemoteIndexCacheTest.scala

```diff
@@ -561,6 +561,108 @@ class RemoteIndexCacheTest {
+    assertCacheSize(2)
+    val missingMetadata = metadataList(0)
+    val missingEntry = entry0
+    // wait until evicted entry is marked for deletion
+    TestUtils.waitUntilTrue(() => missingEntry.isMarkedForCleanup,
+      "Failed to mark evicted cache entry for cleanup after resizing cache.")
+    TestUtils.waitUntilTrue(() => missingEntry.isCleanStarted,
+      "Failed to cleanup evicted cache entry after resizing cache.")
+    // verify no index files for `missingEntry` on remote cache dir
+    TestUtils.waitUntilTrue(() => !getIndexFileFromRemoteCacheDir(remoteOffsetIndexFileName(missingMetadata)).isPresent,
+      s"Offset index file for evicted entry should not be present on disk at ${cache.cacheDir()}")
+    TestUtils.waitUntilTrue(() => !getIndexFileFromRemoteCacheDir(remoteTimeIndexFileName(missingMetadata)).isPresent,
+      s"Time index file for evicted entry should not be present on disk at ${cache.cacheDir()}")
+    TestUtils.waitUntilTrue(() => !getIndexFileFromRemoteCacheDir(remoteTransactionIndexFileName(missingMetadata)).isPresent,
+      s"Txn index file for evicted entry should not be present on disk at ${cache.cacheDir()}")
+    TestUtils.waitUntilTrue(() => !getIndexFileFromRemoteCacheDir(remoteDeletedSuffixIndexFileName(missingMetadata)).isPresent,
+      s"Index file marked for deletion for evicted entry should not be present on disk at ${cache.cacheDir()}")
+
+    // Reduce cache capacity to only store 1 entries
+    cache.resizeCacheSize(1 * estimateEntryBytesSize)
+    assertCacheSize(1)
+
+    val nextMissingMetadata = metadataList(1)
+    val nextMissingEntry = entry1
+    // wait until evicted entry is marked for deletion
+    TestUtils.waitUntilTrue(() => nextMissingEntry.isMarkedForCleanup,
+      "Failed to mark evicted cache entry for cleanup after resizing cache.")
+    TestUtils.waitUntilTrue(() => nextMissingEntry.isCleanStarted,
+      "Failed to cleanup evicted cache entry after resizing cache.")
+    // verify no index files for `nextMissingEntry` on remote cache dir
+    TestUtils.waitUntilTrue(() => !getIndexFileFromRemoteCacheDir(remoteOffsetIndexFileName(nextMissingMetadata)).isPresent,
+      s"Offset index file for evicted entry should not be present on disk at ${cache.cacheDir()}")
+    TestUtils.waitUntilTrue(() => !getIndexFileFromRemoteCacheDir(remoteTimeIndexFileName(nextMissingMetadata)).isPresent,
+      s"Time index file for evicted entry should not be present on disk at ${cache.cacheDir()}")
+    TestUtils.waitUntilTrue(() => !getIndexFileFromRemoteCacheDir(remoteTransactionIndexFileName(nextMissingMetadata)).isPresent,
+      s"Txn index file for evicted entry should not be present on disk at ${cache.cacheDir()}")
+    TestUtils.waitUntilTrue(() => !getIndexFileFromRemoteCacheDir(remoteDeletedSuffixIndexFileName(nextMissingMetadata)).isPresent,
+      s"Index file marked for deletion for evicted entry should not be present on disk at ${cache.cacheDir()}")
```

Comment:
These lines are duplicated, please extract them as a method (function scope method is enough). Maybe named: `verifyEntryIsEvicted`?

Reply:
So, maybe: 

// resize to the same size, all entries should be kept
cache.resizeCacheSize(1 * estimateEntryBytesSize)

// verify all existing entries are kept

// increase the size
cache.resizeCacheSize(2 * estimateEntryBytesSize)

// verify all existing entries are kept

Reply:
updated.